Halos/HaloModels/TNFW.py

```diff
@@ -0,0 +1,276 @@
+                self._zeval = self.z
+            else:
+                self._zeval = self.z_infall
+
+        return self._zeval
+
+    @property
+    def params_physical(self):
+        """
+        See documentation in base class (Halos/halo_base.py)
+        """
+        if not hasattr(self, '_params_physical'):
+            [concentration, rt] = self.profile_args
+            rhos, rs, r200 = self._lens_cosmo.NFW_params_physical(self.mass, concentration, self.z)
+            self._params_physical = {'rhos': rhos, 'rs': rs, 'r200': r200, 'r_trunc_kpc': rt}
+
+        return self._params_physical
+
+    @property
+    def profile_args(self):
+        """
+        See documentation in base class (Halos/halo_base.py)
+        """
+        if not hasattr(self, '_profile_args'):
+
+            truncation_radius = self._lens_cosmo.truncation_roche(self.mass, self.r3d,
+                                                                  self._args['RocheNorm'], self._args['RocheNu'])
+
+            self._profile_args = (self.c, truncation_radius)
+
+        return self._profile_args
+
+# Adding code to be compatible with emulator data
+def tnfw_mass_fraction(tau, c):
+    """
+    This function returns the fraction = final_mass/initial_mass, assuming a truncated NFW profile
+    :param tau: the truncation radius in units of the scale radius 
+    :param c: the halo concentration
+    """
+    x = c
+    Rs = 1.0
+    r_trunc = tau * Rs
+    func = (r_trunc ** 2 * (-2 * x * (1 + r_trunc ** 2) + 4 * (1 + x) * r_trunc * np.arctan(x / r_trunc) -
+                            2 * (1 + x) * (-1 + r_trunc ** 2) * np.log(Rs) + 2 * (1 + x) * (-1 + r_trunc ** 2) * np.log(Rs * (1 + x)) +
+                            2 * (1 + x) * (-1 + r_trunc ** 2) * np.log(Rs * r_trunc) -
+                            (1 + x) * (-1 + r_trunc ** 2) * np.log(Rs ** 2 * (x ** 2 + r_trunc ** 2)))) / (2. * (1 + x) * (1 + r_trunc ** 2) ** 2)
+    mass_loss = func / (np.log(1+c)-c/(1+c))
+    return mass_loss
+    
+def tau_mf_interpolator():
+
+    N = 250
+    tau = np.logspace(-3.5, 2.5, N)
+    concentration = np.linspace(1.0, 200.0, N)
+    
+    log10_mass_fraction_1d = np.linspace(-4, -0.001, N)
+    log10tau_2d = np.zeros((N, N))
+
+    # This computes the value of tau that correponds to each pair of (concentration, mass_loss) 
+    for i, con_i in enumerate(concentration):
+        mfinal = tnfw_mass_fraction(tau, con_i)
+        log10final_mass = np.log10(mfinal)
+        mfinterp = interp1d(log10final_mass, np.log10(tau), fill_value='extrapolate')
+        
+        for j, log10_mass_j in enumerate(log10_mass_fraction_1d):
+            log10tau_2d[i,j]  = float(mfinterp(log10_mass_j))
+
+    interp_points = (concentration, log10_mass_fraction_1d)
+    interpolator = RegularGridInterpolator(interp_points, log10tau_2d, fill_value=None, bounds_error=False)
+    return interpolator
+
+truncation_radius_interpolator = tau_mf_interpolator()
```

Comment:
since this function is only going to be called inside the class it should be private, i.e. it should start with an underscore _truncation_radius_interpolator. This will prevent people from importing it by mistake